applications/FluidDynamicsApplication/custom_elements/alternative_d_vms_dem_coupled.cpp

```diff
@@ -798,40 +1477,187 @@ void AlternativeDVMSDEMCoupled<TElementData>::CalculateStabilizationParameters(
+
+    for (unsigned int i = 0; i < NumNodes; i++)
+        for (unsigned int d = 0; d < Dim; d++)
+            fluid_fraction_gradient[d] += rData.DN_DX(i,d) * rData.FluidFraction[i];
+
 
-    double det_permeability = MathUtils<double>::Det(permeability);
-    MathUtils<double>::InvertMatrix(permeability, sigma, det_permeability, -1.0);
+    // This last term does not exist physically and it is included to do the spectral radius taking into account the inverse Gamma
+    // whose size is (d+1,d+1)
 
     double velocity_modulus = 0.0;
     double fluid_fraction_gradient_modulus = 0.0;
-    double sigma_term = 0.0;
     for (unsigned int d = 0; d < Dim; d++){
         velocity_modulus += Velocity[d] * Velocity[d];
         fluid_fraction_gradient_modulus += std::pow(fluid_fraction_gradient[d],2);
-        for (unsigned int e = d; e < Dim; e++){
-            sigma_term += std::pow(sigma(d,e),2);
-        }
     }
 
     double velocity_norm = std::sqrt(velocity_modulus);
     double fluid_fraction_gradient_norm = std::sqrt(fluid_fraction_gradient_modulus);
+
     double c_alpha = fluid_fraction + h / c1 * fluid_fraction_gradient_norm;
-    inv_tau = (c1 * viscosity / (h*h) + density * (c2 * velocity_norm / h ) ) * c_alpha + density * fluid_fraction / rData.DeltaTime + std::sqrt(sigma_term);
-    inv_tau_NS = c1 * viscosity / (h*h) + density * (c2 * velocity_norm / h ) + std::sqrt(sigma_term);
-    double tau_one = 1 / inv_tau;
-    double tau_one_NS = 1 / inv_tau_NS;
+
+    inv_tau_NS = c1 * viscosity / std::pow(h/(p*p),2.0) + density * (c2 * velocity_norm / (h/p) );
+    double tau_one_NS = 1.0 / inv_tau_NS;
+
+    //this->CalculateSpectralRadius(rData, spectral_radius, tau_one_NS, c1, sigma);
+
+    inv_tau = (c1 * viscosity / std::pow(h/(p*p),2.0) + density * (c2 * velocity_norm / (h/p) ) ) * c_alpha + density * fluid_fraction / rData.DeltaTime;
+    double tau_one = 1.0 / (inv_tau + sigma(0,0));
 
     TauOne = tau_one * I;
-    TauTwo = h * h / (c1 * fluid_fraction * tau_one_NS);
+    TauTwo = std::pow(h/p,2.0) / (c1 * fluid_fraction * tau_one_NS);
+}
+
+template< class TElementData >
+void AlternativeDVMSDEMCoupled<TElementData>::CalculateSpectralRadius(
+    const TElementData& rData,
+    double& spectral_radius,
+    double tau_one_NS,
+    const double c1,
+    MatrixType matrix) const
+{
+    const double h = rData.ElementSize;
+    MatrixType inv_Gamma = IdentityMatrix(Dim+1, Dim+1);
+    MatrixType eigen_val_mat, resulting_mat;
+    MatrixType eigen_vect_mat = IdentityMatrix(Dim+1, Dim+1);
+
+    double ji = h*h /(c1*std::pow(tau_one_NS,2));
+
+    inv_Gamma(Dim,Dim) += 1/ji;
+
+    resulting_mat = prod(inv_Gamma, eigen_vect_mat);
+
+    this->GaussSeidelEigenSystem(matrix, resulting_mat, eigen_val_mat);
+
+    for(unsigned int d = 0; d <= Dim; d++)
+        for(unsigned int e = 0; e <= Dim; e++)
+            spectral_radius = 0*std::max(eigen_val_mat(d,e), spectral_radius);
```

Comment:
`0*..` is not always 0?

Reply:
This is an error, this function has to be removed. It is not used.